test/helpers/kubectl.go

```diff
@@ -1868,12 +1868,15 @@ func (kub *Kubectl) CiliumExecContext(ctx context.Context, pod string, cmd strin
 	return res
 }
 
-// CiliumExec runs cmd in the specified Cilium pod.
-// Deprecated: use CiliumExecContext instead
-func (kub *Kubectl) CiliumExec(pod string, cmd string) *CmdRes {
-	ctx, cancel := context.WithTimeout(context.Background(), HelperTimeout)
-	defer cancel()
-	return kub.CiliumExecContext(ctx, pod, cmd)
+// CiliumExecMustSucceed runs cmd in the specified Cilium pod.
+// it causes a test failure if the command was not successful.
+func (kub *Kubectl) CiliumExecMustSucceed(ctx context.Context, pod, cmd string, optionalDescription ...interface{}) *CmdRes {
+	res := kub.CiliumExecContext(ctx, pod, cmd)
+	if !res.WasSuccessful() {
+		res.SendToLog(false)
+	}
+	res.ExpectSuccess(optionalDescription...)
```

Comment:
I think that currently this will report in the Ginkgo output that the failure is on this, line 1878 of the `test/helpers/kubectl.go`, which is not super useful since it doesn't point where in the actual test the failure occurred.

There's this `ExpectWithOffset(...)` variation on the expect functions which allows you to pass a parameter specifying how many layers deep the current helper is. I think that `CiliumExecMustSucceed` would usually be used somewhere in a regular test, so we can probably switch it to `ExpectWithOffset(1, ...)`  so that ginkgo, upon failure, will report the failure line in the caller rather than in this file.